pkg/runtime/node.go

```diff
@@ -0,0 +1,37 @@
+package runtime
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+// Node represents a configured node with everything needed  to connect to the given host
+// which is defined in the type property
+// If the type is not available the test will fail and stop its execution
+type Node struct {
+	Name           string
+	Type           string
+	User           string
+	Pass           string
+	Addr           string
+	Image          string
+	IdentityFile   string
+	Privileged     bool
+	DockerExecUser string
+}
+
+func (n *Node) ExpandEnv() {
+	n.Name = os.ExpandEnv(n.Name)
+	n.Type = os.ExpandEnv(n.Type)
+	n.User = os.ExpandEnv(n.User)
+	n.Addr = os.ExpandEnv(n.Addr)
+	n.Image = os.ExpandEnv(n.Image)
+	n.IdentityFile = os.ExpandEnv(n.IdentityFile)
+	n.DockerExecUser = os.ExpandEnv(n.DockerExecUser)
+
+	if n.Pass != "" && !strings.Contains(n.Pass, "$") {
```

Comment:
I like this a lot. I assume if node types begin to expand, i.e more feature are added to docker. This would be the place to implement it?

Reply:
Yes, if a property should support env variables it needs to go in here.
I didn't implemented a constructor function like `func NewNode() *Node` because it would need every property as an input argument and that would lead to annoying refactorings if a property is added.